### PR TITLE
feat: support api key on here provider

### LINKED ISSUE #1041: Here Provider now uses an API Key for REST
Would it be possible to get support for the API KEY authentication in the Here provider? It uses a different URL and only needs the API KEY attribute.

### PR DESCRIPTION
**Pull request by atymic**
I've updated the provider so it can use both the API key and appcode/appkey.
The issue is that's a bit of a mess. Since the appcode/appkey is deprecated, maybe we should consider bumping the major version and dropping support for the old style auth?

Closes #1041

### CODE REVIEW

**Comment by jbelien** [comment]
Thanks @atymic ! 👍 
I'll review it but could you make sure it passes the tests first.

AppCode doesn't seem to be completely deprecated, they are still usable but new users apparently can't create it anymore (see https://developer.here.com/documentation/authentication/dev_guide/topics/app-credentials.html)
Therefore I would prefer to keep both options possible.
What's your opinion @Nyholm ?

**Comment by atymic** [comment]
> I'll review it but could you make sure it passes the tests first.

They actually do all pass locally, something is wonky with CI.
Will try to figure it out

> AppCode doesn't seem to be completely deprecated, they are still usable but new users apparently can't create it anymore (see developer.here.com/documentation/authentication/dev_guide/topics/app-credentials.html)
Therefore I would prefer to keep both options possible.

While I did implement it with both options, on second thoughts I don't think there's much point it having both. Anyone still on the old auth method can switch over via the dashboard pretty easily, and nothing is going to break for existing users (due to the major version bump)

Anyone new coming to the package can use the next major version with the API key.

**Comment by atymic** [comment]
Ok, i've tried redoing the cache files with no luck. Any ideas why they are not working in CI?

image

**Comment by alteredorange** [comment]
So is Here not currently working? I just signed up, and can't get app code credentials, but when I just put in an API key, it says I need an app code...

**Comment by pfuhrmann** [comment]
Would be great to get this merged. The current state makes `here-provider` unusable for any new user or any user that needs to refresh the credentials.

@jbelien do you have ideas about why CI is failing? Can I help somehow to get this merged @atymic ?

**Comment by jbelien** [comment]
> @jbelien do you have ideas about why CI is failing? Can I help somehow to get this merged @atymic ?

I'm having the same issue locally ... Investigating it now :)

File: src/Provider/Here/Readme.md [hunk 1]
```diff
@@ -8,23 +8,26 @@
...
 
 ```bash
 composer require geocoder-php/here-provider
 ```
 
-### App Id and App Code
+## Using
 
-Get your Here credentials at https://developer.here.com/
+TODO
```
Comment (jbelien) [inline src/Provider/Here/Readme.md:24]: Add documentation on how to use it (API key vs AppId/AppCode) !

**Comment by jbelien** [comment]
@atymic Could you add some documentation and merge the Style CI fix we did in `master` ? Thanks.

**Comment by atymic** [comment]
On it :)

**Comment by atymic** [comment]
@jbelien docs added and style issues resolved :)

**Comment by jbelien** [comment]
Awesome! Thanks @atymic 🎉

**Comment by jbelien** [comment]
Version `0.5.0` is [released](https://packagist.org/packages/geocoder-php/here-provider).

**Pull request merged**